packages/build-tools/src/steps/functions/repack.ts

```diff
@@ -0,0 +1,187 @@
+import fs from 'node:fs';
+import os from 'node:os';
+import path from 'node:path';
+
+import { type Job, Platform } from '@expo/eas-build-job';
+import { type bunyan } from '@expo/logger';
+import {
+  BuildFunction,
+  BuildStepInput,
+  BuildStepInputValueTypeName,
+  BuildStepOutput,
+  spawnAsync,
+} from '@expo/steps';
+import {
+  repackAppAndroidAsync,
+  repackAppIosAsync,
+  type Logger,
+  type SpawnProcessAsync,
+  type SpawnProcessOptions,
+  type SpawnProcessPromise,
+  type SpawnProcessResult,
+} from '@expo/repack-app';
+import { v4 as uuidv4 } from 'uuid';
+
+import { pathExistsAsync } from '../../utils/files';
+
+export function createRepackBuildFunction(): BuildFunction {
+  return new BuildFunction({
+    namespace: 'eas',
+    id: 'repack',
+    name: 'Repack app',
+    inputProviders: [
+      BuildStepInput.createProvider({
+        id: 'source_app',
+        allowedValueTypeName: BuildStepInputValueTypeName.STRING,
+        required: true,
+      }),
+      BuildStepInput.createProvider({
+        id: 'output_path',
+        allowedValueTypeName: BuildStepInputValueTypeName.STRING,
+        required: false,
+      }),
+      BuildStepInput.createProvider({
+        id: 'working_directory',
+        allowedValueTypeName: BuildStepInputValueTypeName.STRING,
+        required: false,
+      }),
+      BuildStepInput.createProvider({
+        id: 'embed_bundle_assets',
+        allowedValueTypeName: BuildStepInputValueTypeName.BOOLEAN,
+        required: false,
+      }),
+    ],
+    outputProviders: [
+      BuildStepOutput.createProvider({
+        id: 'output_path',
+        required: true,
+      }),
+    ],
+    fn: async (stepsCtx, { inputs, outputs, env }) => {
+      const projectRoot = stepsCtx.workingDirectory;
+      const verbose = stepsCtx.global.env['EAS_VERBOSE'] === '1';
+      const repackLogger = createBunyanLoggerAdapter(stepsCtx.logger);
+      const repackSpawnAsync = createSpawnAsyncStepAdapter(verbose, stepsCtx.logger);
+
+      const tmpDir = path.join(os.tmpdir(), `repack-${uuidv4()}`);
```

Comment:
For temporary directories I've seen us use `mkdtemp` instead of `uuidv4`

https://github.com/expo/eas-build/blob/e4af829932233d84ed0d9aa430a201aeaa667db5/packages/build-tools/src/steps/functions/downloadBuild.ts#L121-L123